packages/storage/exp/api.ts

```diff
@@ -255,10 +256,14 @@ export function ref(
   serviceOrRef: StorageService | StorageReference,
   pathOrUrl?: string
 ): StorageReference | null {
-  return refInternal(
-    serviceOrRef as StorageServiceInternal | Reference,
-    pathOrUrl
-  );
+  if (serviceOrRef instanceof Reference) {
+    return refInternal(serviceOrRef as Reference, pathOrUrl);
+  } else {
+    return refInternal(
+      getModularInstance(serviceOrRef) as StorageServiceInternal,
+      pathOrUrl
+    );
+  }
```

Comment:
We don't need to check whether it's a `StorageService` or `Reference`. Both of them can be a compat instance, we can just call `getModularInstance()` and pass the result to `refInternal()`.

Reply:
Updated.